### DISCUSSION TITLE
Are there plans to change-to/support USWDS 3.0?

### DISCUSSION THREAD

**Discussion by kylebayler** [trussworks/react-uswds#2176] [Ideas]
USWDS v3.0 was released on 28th April 2022, and I was wondering if there are any plans to either change to or support this within this package? Or whether this package will support USWDS v2.0 only. 

https://designsystem.digital.gov/about/releases/
https://designsystem.digital.gov/whats-new/updates/2022/04/28/introducing-uswds-3-0/

**Comment by ebhbahfema** [comment]
Bump!

**Comment by rogeruiz** [comment]
Thank you @kbayliss for your question. At this time, the project does not have a roadmap for when we will be able to release a version of React-USWDS that supports USWDS 4.0. This is for a number of reasons, but mainly we will need to ensure compatibility with our React components with the new structure of USWDS components. Even writing this, I realize how overloaded the work components can sound in this project.

Our current plan that we discussed internally when the original announcement happened was to create an issues backlog since this feels like a major upgrade. This change will introduce React-USWDS 4.0 as we feel that it can possibly have breaking changes for users of React-USWDS 3.x.x. 

I've added some labels to this discussion @kbayliss for both our internal team and for outside contributors such as yourself to communicate the level of effort and changes that would need to be made to track this sort of upgrade. If you would like to contribute to this effort, let's have discussions in this thread to generate GitHub issues that will help track this work. I want to post this now but I will come back today to post a follow up of issues that I think would be good to create and work on and how we can map those to pull requests.

**Reply by rogeruiz** [reply]
Thanks again for your patience with us @kbayliss. So far we've come up with some strategies that can be used to help define some tickets. We'll be looking through this more on our end so there's no pressure for you or anyone else to create these issues but we'd like to encourage folks to contribute in other ways besides just writing code.

1. We must consider changes to our Sass files. We currently would like this work tracked in smaller Issues or Pull Requests if possible. The project also currently uses `uswds: 2.13.3` so [this seems like it may be something that would not take too long](https://designsystem.digital.gov/documentation/migration/).
2. It seems like we have a deadline of migrating away from `uswds: 2.13.3` and onto `uswds: 3.x` here by October of 2022, if we'd like to stay current. We're unsure right now what that means for us since we haven't made the change. But we're sure to track any breaking changes moving forward.
3. We have component changes that will need to be accounted for. This means that we would like to keep Issues and Pull Requests as small as possible when working through them. This will help us manage and track the workload.
4. We have a documentation problem in the project as we currently hard-code the version 2.0 in various places within the code base and the documentation for the code base. This would be a great low-lift solution that we could see OSS contributors members taking on by submitting either Issues or Pull Requests.

All of the above said, our internal maintainers team is aware of some of these deadlines and things we need to update. We will be monitoring this Discussion thread more closely. If you subscribe to the repository, you should get GitHub notifications when the Issues or Pull Requests are created. And as always, Pull Requests and Issues are always welcome from outside contributors.

Thanks again.

**Reply by aligg** [reply]
Hi! I missed this discussion until now and had previously filed a related issue: https://github.com/trussworks/react-uswds/issues/2199

@rogeruiz it sounds like we're aiming to upgrade to `uswds: 3.x` this month? Is that still the case? I'm also someone whose hoping for this so we can use react-uswds while on a uswds v3 version. 

It sounds like the team is expecting this to be a heavier lift than the quickness USDS promises in the docs you linked to, potentially with subtasks around updating documentation, upgrading components, upgrading Sass files and the actual upgrade itself? Happy to create separate tickets for that if it would be helpful.

**Reply by rogeruiz** [reply]
Thanks @aligg yes your summary matches our understanding on our team. We encourage opening either Issues or PRs following our ~~repositories~~ repository's contributions conventions. Thank you again. 🎉

**Reply by aligg** [reply]
Noting there is now a PR for this 🎉 (Thanks @kleinschmidtj): [kleinschmidtj](https://github.com/trussworks/react-uswds/pull/2203)
